packages/grid/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterCell.tsx

```diff
@@ -192,8 +194,8 @@ const GridHeaderFilterCell = React.forwardRef<HTMLDivElement, GridHeaderFilterCe
     const onMouseDown = React.useCallback(
       (event: React.MouseEvent) => {
         if (!hasFocus) {
-          if (inputRef.current && inputRef.current.contains(event.target as HTMLElement)) {
-            inputRef.current.focus();
+          if (inputRef.current && inputRef.current.contains?.(event.target as HTMLElement)) {
+            inputRef.current?.focus();
```

Comment:
I am guessing this won't be needed if the reference is not `undefined` and it includes the `contains` function too 🤔 
```suggestion
            inputRef.current.focus();
```

Reply:
Right!